spring-web/src/main/java/org/springframework/http/MediaType.java

```diff
@@ -552,14 +552,60 @@ public static MediaType parseMediaType(String mediaType) {
 		if (!StringUtils.hasLength(mediaTypes)) {
 			return Collections.emptyList();
 		}
-		String[] tokens = StringUtils.tokenizeToStringArray(mediaTypes, ",");
+		String[] tokens = isAnyCommaDoubleQuotedInString(mediaTypes) ?
```

Comment:
I've tried to be here as descriptive as possible with the method names so the reader can understand the intent before delving into the code